app/code/Magento/ReviewGraphQl/Model/Resolver/Product/Review/AverageRating.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\ReviewGraphQl\Model\Resolver\Product\Review;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Review\Model\RatingFactory;
+use Magento\Review\Model\Review;
+
+/**
+ * Review average rating resolver
+ */
+class AverageRating implements ResolverInterface
+{
+    /**
+     * @var RatingFactory
+     */
+    private $ratingFactory;
+
+    /**
+     * @param RatingFactory $ratingFactory
+     */
+    public function __construct(
+        RatingFactory $ratingFactory
+    ) {
+        $this->ratingFactory = $ratingFactory;
+    }
+
+    /**
+     * Resolves review average rating
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     *
+     * @return float|Value|mixed
+     *
+     * @throws GraphQlInputException
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($value['model'])) {
+            throw new GraphQlInputException(__('Value must contain "model" property.'));
+        }
+
+        /** @var Review $review */
+        $review = $value['model'];
+        $summary = $this->ratingFactory->create()->getReviewSummary($review->getId());
+        $averageRating = $summary->getSum();
+
```

Comment:
What if the $averageRating is null? I ran into such a case and hence querying for customer reviews returned this error.
`"Cannot return null for non-nullable field ProductReview.average_rating."`

Reply:
Looks like @prabhuram93 commented about this case in here.
https://github.com/magento/magento2/pull/27882#discussion_r444650628

Reply:
Just added an improvement to fix this one.
Thank you.